users/serializers.py

```diff
@@ -16,3 +16,21 @@ class Meta:
             'image',
         )
     read_only_fields = ('image', 'last_name')
+
+
+class LoginSerializer(serializers.Serializer):
+    email = serializers.CharField()
+    password = serializers.CharField(
+        style={'input_type': 'password'}
+        )
+
+    def validate(self, *args):
```

Comment:
```
def validate(self, data):
    """ put some descriptive docstrings
    """
    user = authenticate(**data)
    if not user:
        raise serializers.ValidationError("Invalid Email or Password.", code="invalid")
    return data
```

Reply:
also put the user object in a class attribute/property so that you can use it to actually login the user (save the instance in the session). what you did so far was to just check if the user credential is valid.